### PR TITLE
Patch 1 2

### PR DESCRIPTION
**Pull request by aot29**
# MaRDI Pull Request

Issues: #1 , #2 

**Changes**:
- Added action on a schedule, rebuild at midnight every day
- All extensions are now cloned from Github. Most are cloned from the official Mediawiki Github repo, except for TwitterWidget and WikibaseLocalMedia (which are not part of an official release), and MaRDI4NFDI/WikibaseImport, which is custom.
-  All extensions are now from branch 1.35, except for TwitterWidget, which only has the main branch, and MaRDI4NFDI/WikibaseImport, which is custom.

**Instructions for PR review**:
- [ ] Conceptual Review (Logic etc...) 
- [x] Code Review (Review your implementation) 
- [ ] Checkout (Test changes locally) 

**Checklist for this PR**: 
- [ ] [Reviewers and Assignee specified.](https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users)
- [ ] [All related issues are linked.](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 name: docker-wikibase-actions-workflow
-on: 
-  push:
-    branches: 
-      - 'main'
+
+on:
```
Comment (physikerwelt) [inline .github/workflows/main.yml:3]: I think it would be better to have both. Schedule and push / pull request.

File: Dockerfile [hunk 2]
```diff
@@ -8,61 +8,43 @@ RUN apt-get update && \
...
+bash clone-extension.sh MathSearch ${BRANCH};
+
+# clone extensions not officially distributed by mediawiki
+RUN git clone https://github.com/ProfessionalWiki/WikibaseLocalMedia.git -b ${BRANCH} WikibaseLocalMedia &&\
+rm -rf WikibaseLocalMedia/.git
+
+RUN git clone https://github.com/ciencia/mediawiki-extensions-TwitterWidget.git TwitterWidget &&\
+rm -rf TwitterWidget/.git
+
+# clone WikibaseImport from the mardi fork (no branch needed here, as extension is custom made for the portal)
```
Comment (physikerwelt) [inline Dockerfile:44]: If we plan a change from one version to another, it makes sense to have different branches for some time.

File: clone-extension.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+EXTENSION=$1
+BRANCH=$2
+
+git clone https://github.com/wikimedia/mediawiki-extensions-${EXTENSION}.git -b ${BRANCH} ${EXTENSION}
```
Comment (physikerwelt) [inline clone-extension.sh:5]: If we run into problems (speed or resource limit), we can still clone from the official repo

https://gerrit.wikimedia.org/r/mediawiki/extensions/$EXTENSiON

**Review by physikerwelt — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (aot29) [reply Dockerfile:44]: TwitterWidget has only the main branch.

File: clone-extension.sh (hunk 3 above)
Comment (aot29) [reply clone-extension.sh:5]: Yes we could do: If clone from git doesn't work -> fallback on gerrit. But I think it's super unlikely. The worst that could happen is that we are stuck with yesterdays image, that wouldn't be so bad.

File: .github/workflows/main.yml (hunk 1 above)
Comment (aot29) [reply .github/workflows/main.yml:3]: added build on push to main

File: clone-extension.sh (hunk 3 above)
Comment (physikerwelt) [reply clone-extension.sh:5]: I mean gerrit instead of github

Comment (aot29) [reply clone-extension.sh]: Now cloning from Gerrit by default, fallback on Github.

**Pull request merged**